packages/shift-next/src/lib/api-client.js

```diff
@@ -18,6 +18,12 @@ class ApiClient {
     this.client = axios.create({
       baseURL: Config.get().apiHostProxy
     })
+
+    if (options.postRequestHook) {
+      this.postRequestHook = options.postRequestHook
+    } else {
+      this.postRequestHook = () => {}
+    }
```

Comment:
This is where the magic happens - any callable passed in here gets called after the response.  We use this to pass in something that mutates the express request (req) without api-client ever being aware of it